ads/feature_store/dataset.py

```
@@ -518,7 +519,18 @@ def add_models(self, model_details: ModelDetails) -> "Dataset":
             for item in items:
                 model_details.items.append(item)
         self.with_model_details(model_details)
-        return self.update()
+        self.update()
+
+        self._update_from_oci_dataset_model(self.oci_dataset)
+        updated_model_details = self.model_details
+        if updated_model_details and updated_model_details.items and model_details:
+            for model_id in model_details.items:
+                if model_id not in updated_model_details["items"]:
```

why this check if the model will get updated once we invoke `self.update()`. also, this will silently fail

---

the property 'get' gets us the dictionary